### DISCUSSION TITLE
Possible regression in disposing sqlite connections between 1.4.0 and 2.00

### DISCUSSION THREAD

**Discussion by karlicoss** [sqlalchemy/sqlalchemy#9167] [Usage Questions] [answered]
TLDR: seems that in version 2.0.0, sqlite engine isn't necessarily disposing of the connection on `.close()` -- as a result sometimes you can observe wal/shm files where you wouldn't expect to

```
from pathlib import Path
from tempfile import TemporaryDirectory
from sqlalchemy import create_engine, MetaData, Table, Column, String, event


with TemporaryDirectory() as td:
    tdir = Path(td)

    engine = create_engine(f"sqlite:///{tdir}/test.db")
    meta = MetaData()
    table = Table('testtable', meta, Column('testcolumn', String))


    # enable WAL mode
    @event.listens_for(engine, 'connect')
    def enable_wal(dbapi_connection, connection_record):
        dbapi_connection.execute('PRAGMA journal_mode=WAL')

    wal = Path(tdir / 'test.db-wal')

    conn = engine.connect()
    with conn.begin():
        table.create(conn, checkfirst=True)
        conn.execute(table.insert(), {'testcolumn': 'value'})

    assert wal.exists()

    conn.close()

    assert not wal.exists()   # .close should dispose of the connection and WAL as well
```

Prior to `2.0.0`, this was working as expected, assert was passing.

After updating to `2.0.0`, this started triggering the assert. I bisected it down to this commit, but not familiar enough with internals to say if it's expected or not https://github.com/sqlalchemy/sqlalchemy/commit/22ed657827b487df9012def07271aed01bd4ae12

I ended up adding `engine.dispose()` as well after `conn.close()` and this fixed the issue for me. However I'm not sure if it's expected, or whether it's a regression so leaving it here just in case

**Comment by CaselIT** [comment, marked as answer]
Hi,

That's expected, since on version 2 the default pool for sqlite was changed to queue pool, so the connection is kept open.

Adding `engine.dispose` is absolutely the correct way. An alternative is to set the pool to the NullPoll as it was in the previous versions
